server/services/CsvGeneratorService v2.js

```diff
@@ -0,0 +1,787 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/* eslint-disable camelcase */
+const moment        = require('moment');
+const converter     = require('json-2-csv');
+const config        = require('../utils/constants');
+const INDEXNAME     = config.INDEXNAME;
+const CSV_BY_USERS  = config.CSV_BY_USERS;
+const MaxCSVRows    = config.MAXCSVROWS;
+const esb           = require('elastic-builder');
+
+export default class CsvGeneratorService {
+  constructor(esDriver, esServer) {
+    this.esDriver = esDriver;
+    this.esServer = esServer;
+  }
+  checkIndexExist = async _req => {
+    try {
+      const { callWithRequest } = this.esDriver.getCluster('data');
+      const indexExist          = await callWithRequest(_req, 'indices.exists', { index: INDEXNAME });
+      return { ok: true, resp: indexExist };
+    } catch (err) {
+      console.error('CSV Generator - CsvGeneratorService - checkIndexExist:', err);
+      return { ok: false, resp: err.message };
+    }
+  };
+  getSingleCSV  = async _req => {
+    const csvId = _req.params.csvId;
+    try {
+      const { callWithRequest } = this.esDriver.getCluster('data');
+      const csv                 = await callWithRequest(_req, 'get', { index: INDEXNAME, id: csvId });
+      const binary              = csv._source.binary;
+      const json                = Buffer.from(binary, 'base64').toString('utf-8');
+      const data                = { filename: csv._source.file, csv: JSON.parse(json) };
+      return { ok: true, resp: data };
+    } catch (err) {
+      console.error('CSV Generator - CsvGeneratorService - getSingleCSV:', err);
+      return { ok: false, resp: err.message };
+    }
+  };
+  saveCSVToIndex = async (_req, file, status, binary, message, username, type) => {
+    try {
+      const { callWithRequest } = this.esDriver.getCluster('data');
+      const date                = moment().format('DD-MM-YYYY HH:mm:ss');
```

Comment:
i'm not very clear about what's the useage for this, if it is for backend, why do we need to format
if we do need a string instead of date object or timestamp, should it be utc?
moment().utc().format()